virttest/utils_misc.py

```diff
@@ -926,9 +926,9 @@ def safe_rmdir(path, timeout=10, session=None):
                 raise
             time.sleep(step)
         except (aexpect.ShellTimeoutError, aexpect.ShellError) as info:
-            raise exceptions.TestSetupError("Failed to remove directory "
-                                            "%s from remote machine: %s "
-                                            % (path, info))
+            raise exceptions.TestSetupFail("Failed to remove directory "
+                                           "%s from remote machine: %s "
+                                           % (path, info))
```

Comment:
> I think there are two way to fix this issue:
> 
>     1. We could implement `TestSetupError` in `avocado/core/exceptions.py`.
> 

For this solution, implement a `TestSetupError` with status `Error`,  should we update the existing `TestSetupFail` that indicate status `Error` now, to make it indicate `Fail` ?   
If not,  i think these 2 function are in a sense duplicate.

But if yes, make `TestSetupFail` indicate status `Fail`,  `TestSetupFail` be called in many other places, if update the meaning of it, we need assess more.

what's your opinion about this ? @YongxueHong 

>     2. like your solution.
> 
> 
> I prefer the first way since we better not touch it but keep it and from my understanding
> it should be a setup error.
> Anyway, both of them are okay for me.

Reply:
> > I think there are two way to fix this issue:
> > ```
> > 1. We could implement `TestSetupError` in `avocado/core/exceptions.py`.
> > ```
> 
> For this solution, implement a `TestSetupError` with status `Error`, should we update the existing `TestSetupFail` that indicate status `Error` now, to make it indicate `Fail` ?
> If not, i think these 2 function are in a sense duplicate.
> 
> But if yes, make `TestSetupFail` indicate status `Fail`, `TestSetupFail` be called in many other places, if update the meaning of it, we need assess more.
> 
> what's your opinion about this ? @YongxueHong
> 
> > ```
> > 2. like your solution.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I prefer the first way since we better not touch it but keep it and from my understanding
> > it should be a setup error.
> > Anyway, both of them are okay for me.

Thanks for explaining,  I review the code about this part, found that I miss some info.
```
class TestSetupFail(TestBaseException):

    """
    Indicates an error during a setup or cleanup procedure.
    """
    status = "ERROR"
```
Actual, it is a type of error. You are right.
Looks good to me.